### PR TITLE
Robert Walz Woche06

### PR DESCRIPTION
**Pull request by LeMixer**
Task 1: 1778099
Task 2: ('/pcqjnl/lrrl/nwjggvr/bwmglvmt'), 1623571)

### CODE REVIEW

**Comment by The-Linux-User** [comment]
Quite a lot of code, especially with the two separate files but seems to work all in all.

File: Exercises/RobertWalz/exercises_06/main.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    return count
+
+
+data = get_data()
+file_system = FileSystem(70000000)
+evaluateTerminalRecord(data=data, fs=file_system)
+dirs: dict[Path, int] = {}
+
+
+get_dir_sizes(file_system.current_dir.parent, dirs)
```
Comment (fabischw) [inline Exercises/RobertWalz/exercises_06/main.py:82]: Does this line server any purpose ? There seems to be a return statement in the function called, however the returned value is not safed in a variable.

File: Exercises/RobertWalz/exercises_06/main.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            sizes_of_dirs[content.path] = content.size
+            dir.size += get_dir_sizes(content, sizes_of_dirs=sizes_of_dirs)
+            sizes_of_dirs[dir.path] = dir.size
+
+    return dir.size
+
+
+def get_size_of_dirs_above_size(size: int, directories: dict[Path, int]):
+    values = [value for value in directories.values() if value <= size]
+    count = sum(values)
```
Comment (fabischw) [inline Exercises/RobertWalz/exercises_06/main.py:72]: Why not sum them immediatly instead of first creating a list and then get the sum ?

**Review by fabischw — APPROVED** [review]
Looks good in general, consider adding some more comments in the future and take a look at the comments I added.

File: Exercises/RobertWalz/exercises_06/main.py (hunk 1 above)
Comment (LeMixer) [reply Exercises/RobertWalz/exercises_06/main.py:82]: Apparently it does. I am guessing it fills the size value for each dict. The return is there for a recursion.

File: Exercises/RobertWalz/exercises_06/main.py (hunk 2 above)
Comment (LeMixer) [reply Exercises/RobertWalz/exercises_06/main.py:72]: Yeah, that's true

**Pull request merged**